src/backend/distributed/commands/alter_table.c

```diff
@@ -1178,7 +1186,7 @@ ReplaceTable(Oid sourceId, Oid targetId, List *justBeforeDropCommands,
 	{
 		changeDependencyFor(RelationRelationId, sequenceOid,
 							RelationRelationId, sourceId, targetId);
-		if (ShouldSyncTableMetadata(sourceId))
+		if (changeSequenceDependenciesOnWorkers && ShouldSyncTableMetadata(sourceId))
```

Comment:
Can we just make this `ShouldSyncTableMetadata (targetId)`?

Reply:
We should know for which table operation we are replacing the table, so I think the `tableId` wouldn't be suffficient itself

Reply:
Sorry, I misunderstood. This makes sense